src/lib/edge-functions/proxy.js

```diff
@@ -141,34 +148,49 @@ const prepareServer = async ({
   port,
   projectDir,
 }) => {
-  const bundler = await import('@netlify/edge-bundler')
-  const distImportMapPath = getPathInProject([DIST_IMPORT_MAP_PATH])
-  const runIsolate = await bundler.serve({
-    ...getDownloadUpdateFunctions(),
-    certificatePath,
-    debug: env.NETLIFY_DENO_DEBUG === 'true',
-    distImportMapPath,
-    formatExportTypeError: (name) =>
-      `${NETLIFYDEVERR} ${chalk.red('Failed')} to load Edge Function ${chalk.yellow(
-        name,
-      )}. The file does not seem to have a function as the default export.`,
-    formatImportError: (name) => `${NETLIFYDEVERR} ${chalk.red('Failed')} to run Edge Function ${chalk.yellow(name)}:`,
-    importMaps,
-    inspectSettings,
-    port,
-  })
-  const registry = new EdgeFunctionsRegistry({
-    bundler,
-    config,
-    configPath,
-    directories,
-    getUpdatedConfig,
-    internalFunctions,
-    projectDir,
-    runIsolate,
-  })
+  try {
+    const bundler = await import('@netlify/edge-bundler')
+    const distImportMapPath = getPathInProject([DIST_IMPORT_MAP_PATH])
+    const runIsolate = await bundler.serve({
+      ...getDownloadUpdateFunctions(),
+      certificatePath,
+      debug: env.NETLIFY_DENO_DEBUG === 'true',
+      distImportMapPath,
+      formatExportTypeError: (name) =>
+        `${NETLIFYDEVERR} ${chalk.red('Failed')} to load Edge Function ${chalk.yellow(
+          name,
+        )}. The file does not seem to have a function as the default export.`,
+      formatImportError: (name) =>
+        `${NETLIFYDEVERR} ${chalk.red('Failed')} to run Edge Function ${chalk.yellow(name)}:`,
+      importMaps,
+      inspectSettings,
+      port,
+    })
+    const registry = new EdgeFunctionsRegistry({
+      bundler,
+      config,
+      configPath,
+      directories,
+      getUpdatedConfig,
+      internalFunctions,
+      projectDir,
+      runIsolate,
+    })
 
-  return { registry, runIsolate }
+    return { registry, runIsolate }
+  } catch (error) {
+    log(
```

Comment:
I think we're assuming that this error will be due to an incompatible platform being used with Deno, but we don't know if that's the case. We're wrapping a fair amount of logic with this `try/catch`, so an error could come from different places, and we risk showing an error message about an unsupported platform to someone running edge functions on a supported platform, which will add to the confusion as opposed to help.

It feels to me that we want to address two separate things:

1. Allow the edge functions initialisation to fail without crashing CLI
2. Showing a useful error message in case people try to use edge functions with an unsupported platform

For point 1, I think a `try/catch` is indeed the right option — although personally I would do it on the `initializeProxy` side and not on `prepareServer`, because as it stands `prepareServer` is a Promise that _resolves_ an error (as opposed to rejecting).

For point 2, I would actually make CLI check the platform being used and output a specific message in that case. This way we can guarantee that we only show that message to people that indeed have an unsupported platform.

Reply:
Point 1: I've moved try/catch to `initializeProxy` but it results in the the error being logged with every request made by cli (screenshot below)

Point 2: I've moved error message to [edge-bundler](https://github.com/netlify/edge-bundler/pull/42) so we only log `error.message`.

<img width="1214" alt="Screenshot 2022-06-14 at 16 30 46" src="https://user-images.githubusercontent.com/16131515/173589481-b146230a-e6ac-4ae6-8bd3-9d87ade1e45b.png">

Reply:
> Point 1: I've moved try/catch to `initializeProxy` but it results in the the error being logged with every request made by cli (screenshot below)

I don't know how you've implemented that, but we should be able to keep track of whether we've already printed the error message or not?

```js
const initializeProxy = async (...) => {
  const server = prepareServer(...)

  let hasServerError = false

  return async (req) => {
    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions || hasServerError) {
      return
    }

    try {
      // Get geolocation and server
    } catch (error) {
      // print error

      hasServerError = true
    }
}
```

This would also avoid calling the geolocation logic on subsequent requests if the server has errored, but it's mostly about respecting the error handling patterns, since `prepareServer` could throw an error when something fails, as opposed to resolving with an error object.

Not a blocker though, so I'll defer to you whether to do this or not.